codec/src/main/java/io/netty/handler/codec/LengthFieldBasedFrameDecoder.java

```diff
@@ -439,10 +442,17 @@ protected Object decode(ChannelHandlerContext ctx, ByteBuf in) throws Exception
         int actualFrameLength = frameLengthInt - initialBytesToStrip;
         ByteBuf frame = extractFrame(ctx, in, readerIndex, actualFrameLength);
         in.readerIndex(readerIndex + actualFrameLength);
-        frameLengthInt = -1; // start processing the next frame
+        resetFrameLengthInt(); // start processing the next frame
         return frame;
     }
 
+    /**
+     * allow repeat read frame
+     */
+    protected void resetFrameLengthInt() {
```

Comment:
Let's remove the `Int` suffix from the method name, since that part is an implementation detail.

I think it'd be good to expand the javadoc to say when/in what situation a subclass might call this method.
@bigT do you have a good suggestion here, perhaps grounded in your use case?

Reply:
@NickleHuang @chrisvest Guys, have you tested the change, is there a unit test that covers this? 

If my reading of the code is correct there is no way to use new API from inside the `getUnadjustedFrameLength`. Say one calls `#resetFrameLengthInt()` inside `getUnadjustedFrameLength(...)`, meaning it returns some frame length and also reset the `this.frameLengthInt` using provided API.

Look at [LengthFieldBasedFrameDecoder.java#L430](https://github.com/netty/netty/blob/14e33dbc2ed85c9965ee6dc3418f69d20bd13546/codec/src/main/java/io/netty/handler/codec/LengthFieldBasedFrameDecoder.java#L430), it looks like the `this.frameLengthInt` is overwritten with the frame length. I think this needs fixing.